src/daft-local-execution/src/lib.rs

```diff
@@ -18,6 +19,36 @@ lazy_static! {
     pub static ref NUM_CPUS: usize = std::thread::available_parallelism().unwrap().get();
 }
 
+/// The `OperatorOutput` enum represents the output of an operator.
+/// It can be either `Ready` or `Pending`.
+/// If the output is `Ready`, the value is immediately available.
+/// If the output is `Pending`, the value is not yet available and a `RuntimeTask` is returned.
+pub(crate) enum OperatorOutput<T> {
+    Ready(T),
+    Pending(RuntimeTask<T>),
+}
```

Comment:
This `OperatorOutput` allows operators to spawn their own tasks on the compute runtime if needed, or if no compute is necessary, return whatever result they have. This also allows hash join probes to await on the probe tables within the `execute` method.